xbout/load.py

```diff
@@ -207,40 +208,43 @@ def _trim(ds, *, guards, keep_boundaries, nxpe, nype):
 
     trimmed_ds = ds.isel(**selection)
 
-    trimmed_ds = trimmed_ds.drop(_BOUT_TIMING_VARIABLES, errors='ignore')
+    trimmed_ds = trimmed_ds.drop(_BOUT_PER_PROC_VARIABLES, errors='ignore')
 
     return trimmed_ds
 
 
 def _infer_contains_boundaries(ds, nxpe, nype):
     """
-    Uses the name of the output file, BOUT++'s topology indices, and the domain
-    decomposition to work out whether this dataset contains boundary cells, and on which
-    side.
-
-    Uses knowledge that BOUT names its output files as /folder/prefix.num.nc,
-    with a numbering scheme
-    num = nxpe*i + j, where i={0, ..., nype}, j={0, ..., nxpe}
+    Uses the processor indices and BOUT++'s topology indices to work out whether this
+    dataset contains boundary cells, and on which side.
     """
 
-    filename = ds.encoding['source']
-
-    *prefix, filenum, extension = Path(filename).suffixes
-    filenum = int(filenum.replace('.', ''))
+    try:
+        xproc = int(ds['PE_XIND'])
+        yproc = int(ds['PE_YIND'])
+    except KeyError:
```

Comment:
Should this behaviour occur silently or throw a warning? (maybe a suppressible one? - there is already an `info` kwarg to `open_boutdataset`)

Reply:
I don't think we need a warning here - both the `try` and the `except` cases are correct: the `try` for BOUT++ versions >=4.3 and the `except` for versions <4.3. We don't gain anything at the moment by using `PE_XIND`/`PE_YIND` instead of the number from the filename, but it should be a bit more future proof in case someone ever decides to change the way processors are distributed (e.g. to collect the y-direction onto a single node for some reason).